test/latex-to-ast.test.ts

```diff
@@ -182,47 +182,53 @@ C`;
     expect(actual.children.length).toBe(1);
     expect(actual.children[0].children[0].type).toBe(ASTNodeTypes.Link);
     expect(actual.children[0].children[0].url).toBe("http://example.com/");
-    expect(actual.children[0].children[0].children[0].value).toBe("http://example.com/");
-  })
+    expect(actual.children[0].children[0].children[0].value).toBe(
+      "http://example.com/"
+    );
+  });
   test("href command", () => {
     const code = `\\href{http://example.com/}{link}`;
     const actual = parse(code);
     expect(actual.children.length).toBe(1);
     expect(actual.children[0].children[0].type).toBe(ASTNodeTypes.Link);
     expect(actual.children[0].children[0].url).toBe("http://example.com/");
     expect(actual.children[0].children[0].children[0].value).toBe("link");
-  })
+  });
   test("label command", () => {
     const code = `\\ref{label}`;
     const actual = parse(code);
     expect(actual.children.length).toBe(1);
     expect(actual.children[0].children[0].type).toBe(ASTNodeTypes.Html);
     expect(actual.children[0].children[0].value).toBe("label");
     expect(actual.children[0].children[0].raw).toBe(code);
-  })
+  });
 });
 
 describe("Fixing document", () => {
   const kernel = new TextlintKernel();
   const options = {
     filePath: "<test>",
     plugins: [
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
```

Comment:
This line is practically right. Hmm..., could you add custom type definitions for 'textlint-rule-*' and remove 'require' syntax to avoid exceptional rules?

Reply:
textlint-rule-spellcheck-tech-word and textlint-rule-ginger do not have type declaration. How do we avoid this?

Reply:
I guess we can write custom type definitions `textlint-rules.d.ts` like

```ts
declare module "textlint-rule-*" { ... };
```

Reply:
I wrote `types/textlint-rule/index.d.ts` as follows.

```ts
import { TextlintRuleModule } from "@textlint/types";

declare module "textlint-rule-*" {
  export default TextlintRuleModule;
}
```

Load it by tsconfig.

```diff
     /* Module Resolution Options */
     // "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
     // "baseUrl": "./",                       /* Base directory to resolve non-absolute module names. */
+    "baseUrl": "./",
     // "paths": {},                           /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
+    "paths": {
+      "textlint-rule-*": ["types/textlint-rule"]
+    },
     // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */
     // "typeRoots": [],                       /* List of folders to include type definitions from. */
     "types": ["node", "jest"],                        /* Type declaration files to be included in compilation. */
```

However, textlint kernel show following error when running `npm run test`.

```
    Definition of rule 'ginger' was not rule module.
    Rule should export function:
    module.exports = function(context){
        // Your rule
    };
```

How do I fix them...

Reply:
Maybe,  


```ts
import  textlintRuleGinger from "textlint-rule-ginger";
import  textlintRuleSpellCheckTechWord from "textlint-rule-spellcheck-tech-word";
```

or

```ts
import * as textlintRuleGinger from "textlint-rule-ginger";
import * as textlintRuleSpellCheckTechWord from "textlint-rule-spellcheck-tech-word";

const textlintRuleGingerDefault = textlintRuleGinger.default;
const textlintRuleSpellCheckTechWordDefault  = textlintRuleSpellCheckTechWord.default;
```

Reply:
It seems to works fine after I fix them as follows.

`types/textlint-rule/index.d.ts`
```ts
import { TextlintRuleReporter } from "@textlint/types";

export const fixer: TextlintRuleReporter;
export const linter: TextlintRuleReporter;
```

`test/latex-to-ast.test.ts`
```ts
import textlintRuleGinger from "textlint-rule-ginger";
import textlintRuleSpellCheckTechWord from "textlint-rule-spellcheck-tech-word";
```

Reply:
It's curious but the running is better than stuck. TypeScript is nice language but TypeScript compiler dances in the wonderland :(